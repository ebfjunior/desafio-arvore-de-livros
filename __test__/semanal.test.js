import React from 'react';

import Semanal from '../src/components/semanal';

describe('Componente Semanal', () => {
  it('Deve renderizar adequadamente', () => {
    const semana = {
        "3" : {"2016-10-12" : 1}
    };
    const wrapper = mount(<Semanal ano="2016" mes="10" periodosSemana={semana}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('Estado deve ser inicializado corretamente', () => {
    const semana = {
        "3" : {"2016-10-12" : 1}
    };
    const wrapper = mount(<Semanal ano="2016" mes="10" periodosSemana={semana}/>);

    expect(wrapper.state().semana).toEqual("3");
    expect(wrapper.state().periodos).toEqual(semana["3"]);
  });

});