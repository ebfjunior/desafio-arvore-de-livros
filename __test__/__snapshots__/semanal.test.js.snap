// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Semanal Deve renderizar adequadamente 1`] = `
<Semanal
  ano="2016"
  mes="10"
  periodosSemana={
    Object {
      "3": Object {
        "2016-10-12": 1,
      },
    }
  }
>
  <span>
    Hello, √Årvore de Livros
  </span>
</Semanal>
`;
